### ISSUE TITLE
[java.lang.IllegalAccessException] - com.appsflyer.internal.AFa1xSDK$$.valueOf

### ISSUE THREAD

**Issue by VinuPolly-Bonnet** [AppsFlyerSDK/appsflyer-android-sdk#48]
I am getting a few crashes due to IllegalAccess Exception from AppsFlyer Internal SDK. Crashlog as seen below:

```
Caused by java.lang.IllegalAccessException: Class java.lang.Class<java.lang.reflect.Method> cannot access protected  method java.lang.Class com.appsflyer.internal.AFa1xSDK$$.loadClass(java.lang.String, boolean) of class java.lang.Class<com.appsflyer.internal.AFa1xSDK$$>
       at java.lang.reflect.Method.invokeImpl(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:411)
       at com.appsflyer.internal.AFa1xSDK$$.valueOf(:100)
       at com.appsflyer.internal.AFa1xSDK$$.AFKeystoreWrapper(:26)
       at java.lang.reflect.Method.invokeImpl(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:411)
       at com.appsflyer.internal.AFa1ySDK.values(:108)
       at com.appsflyer.internal.AFc1xSDK.AFInAppEventType(:142)
       at com.appsflyer.internal.AFd1tSDK.AFKeystoreWrapper(:65)
       at com.appsflyer.internal.AFc1hSDK.valueOf(:113)
       at com.appsflyer.internal.AFc1mSDK$5.run(:109)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)
```

This **issue** seems to be only happening on **Android 8**

### Devices
1. **Motorola One Vision** running on Android 8.1.0
2. **Samsung Galaxy A70** running on Android 8.1.0

**Comment by KevinJ1008** [comment]
Same issue here not most users facing this problem,
Device:  **Vivo S1** Android 8.1.0

**Comment by Turalllb** [comment]
appsflyer              : '6.10.1',
        apsflyerInstallReferrer: '2.2',

one crash per Device: Vivo S1 Android 8.1.0


```Caused by java.lang.IllegalAccessException: Class java.lang.Class<java.lang.reflect.Method> cannot access protected  method java.lang.Class com.appsflyer.internal.AFa1wSDK$$.loadClass(java.lang.String, boolean) of class java.lang.Class<com.appsflyer.internal.AFa1wSDK$$>
       at java.lang.reflect.Method.invokeImpl(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:411)
       at com.appsflyer.internal.AFa1wSDK$$.AFKeystoreWrapper(:100)
       at com.appsflyer.internal.AFa1wSDK$$.AFInAppEventParameterName(:16)
       at java.lang.reflect.Method.invokeImpl(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:411)
       at com.appsflyer.internal.AFa1vSDK.values(:117)
       at com.appsflyer.internal.AFc1pSDK.values(:140)
       at com.appsflyer.internal.AFd1hSDK.AFInAppEventParameterName(:87)
       at com.appsflyer.internal.AFd1nSDK.AFKeystoreWrapper(:114)
       at com.appsflyer.internal.AFd1vSDK$2.run(:109)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)```

**Comment by anyasamoylova** [comment]
The issue is still relevant in version 6.11.1. Found only on android 8. Devices: 

- Samsung Galaxy A70
- Huawei Enjoy 9s
- Vivo S1
- Motorola One Vision

**Comment by pazlavi** [comment]
@anyasamoylova  Please open a support ticket and provide the stack trace.

**Comment by anyasamoylova** [comment]
@pazlavi By opening a support ticket, do you mean creating a new issue?

**Comment by pazlavi** [comment]
@anyasamoylova  contact our support team at support@appsflyer.com as mentioned in the first comment in this thread.

**Comment by roblframpton** [comment]
This is most likely [this issue](https://github.com/AppsFlyerSDK/appsflyer-android-app/issues/20), which for some reason has been raised in the Android sample app repo instead of this one.  Apparently a fix is in progress and due in v6.12.2, but no word on when that will be yet

**Comment by ge-af** [comment]
Hi, 
We've just released a new Android SDK version that fixes this issue.
The new version is v6.12.2.
